engine_elasticsearch.go

```diff
@@ -164,17 +175,17 @@ func (ee elasticEngine) Ids(collection Collection, stopchan chan struct{}) (chan
 
 func (ee elasticEngine) query(collection Collection, q string, closechan chan struct{}) (chan Document, error) {
 	cont := make(chan Document)
-
 	res, err := ee.client.Post(fmt.Sprintf("%s/%s/%s/_search", ee.baseURL, ee.indexName, collection.name), "application/json", strings.NewReader(q))
 	if err != nil {
 		panic(err)
 	}
-
 	switch {
 	case res.StatusCode == 200:
 	case res.StatusCode == 400:
 		res.Body.Close()
 		return nil, ErrInvalidQuery
+	case res.StatusCode == 404:
+		return nil, ErrNotFound
```

Comment:
Can we assume for for this version that it's the same "collection not found" or a "record in the collection not found" - both follow the same query pattern and natively ES returns 404 for either